misc/helper-scripts/rosmsg_to_asn1.py

```diff
@@ -0,0 +1,417 @@
+    parser.add_argument(
+        "-v", "--verbose", action='store_true',
+        help="Verbose output of operation")
+    parser.add_argument(
+        "-l", "--log", action='store_true',
+        help="Log the operation at the default path /tmp/rosmsg_to_asn1.log")
+    parser.add_argument(
+         "--logfile", nargs='?', help="Set path to log the operation.")
+    parser.add_argument(
+        "messages", nargs="*",
+        help="The list of messages to be converted to ASN.1")
+
+    return parser.parse_args()
+
+
+def get_message_package(rospack, msg):
+    """
+    Get the package name a message belongs to. If the message is not found in
+    any of the packages or in multiple packages a message is logged and a None
+    value is returned.
+    :param rospack: A RosPack instance.
+    :param msg: The message to search the package for.
+    :return msg_name: The name of the message (None if not found).
+    :return pkg_name: The name of the package (None if not found).
+    """
+    msg_name = None
+    pkg_name = None
+    # If message is prefixed search in that package only.
+    if "/" in msg:
+        msgs = list(rosmsg.list_msgs(msg.split('/')[0], rospack))
+        if msg not in msgs:
+            logging.error("Couldn't find the message {0}".format(msg))
+        else:
+            pkg_name = msg.split('/')[0]
+            msg_name = msg.split('/')[1]
+    # If there is no prefix find all the messages with that name.
+    # If multiple messages with the same name exist print a message and exit.
+    else:
+        messages = list(rosmsg.rosmsg_search(rospack, rosmsg.MODE_MSG, msg))
+        if len(messages) > 1:
+            s = ""
+            comma = ""
+            for m in messages:
+                s += comma + m
+                comma = ", "
+            logging.error(
+                "Found {0} messages with name {1}. Please clarify the one you"
+                " want by prefixing the package name.\n"
+                "[".format(len(messages), msg) + s + "]")
+        elif len(messages) == 1:
+            logging.debug("Found the message {0}".format(messages[0]))
+            pkg_name = messages[0].split('/')[0]
+            msg_name = messages[0].split('/')[1]
+        else:
+            logging.error("Couldn't find the message {0}".format(msg))
+
+    return msg_name, pkg_name
+
+
+def get_python_msg_module(pkg):
+    """
+    Returns the Python module that includes the of the given package.
+    :param pkg: The name of the package.
+    :return module: The Python module of the package (None if not found).
+    """
+    module_name = pkg + ".msg"
+    module = None
+    try:
+        logging.debug("Trying to import {0}".format(module_name))
+        module = importlib.import_module(module_name)
+    except ImportError as e:
+        logging.error(e.msg)
+        logging.error(
+            "Could not load the package {0}."
+            "Is it in your package path?".format(module_name))
+    return module
```

Comment:
Actually there is no newline escape here. The strings are simply concatenated. I should leave a space in the end of the first one. Did it that way so that it is PEP compliant with 80 chars per line.

Reply:
My point was to join the two logging statements to include the error message in the second one. I added the newline to make a clear break between the error message and the potential solution.

Reply:
Ah ok fixed. :)